Reflection_Engine/Convert/ToFunc.cs

```diff
@@ -0,0 +1,118 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+
+namespace BH.Engine.Reflection
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+
+        public static Func<object[], object> ToFunc(this MethodBase method)
```

Comment:
This set of methods are the port of `CompileFunction` in the BHoM_UI: https://github.com/BHoM/BHoM_UI/blob/d065743b720aa171b442b91de4b07da27974509b/BHoM_UI/Templates/MethodCaller.cs#L144

It has now been refactored into a `Convert` method that returns a ` Func<object[], object>`.
The convergence of any method to a `Delegate` that takes `object[]` as input and returns `object` is necessary to generalise the `MethodCaller` in the UI.